timetypes/rfc3339_value.go

```diff
@@ -76,6 +80,46 @@ func (v RFC3339) StringSemanticEquals(_ context.Context, newValuable basetypes.S
 	return currentRFC3339time.Format(time.RFC3339) == newRFC3339time.Format(time.RFC3339), diags
 }
 
+// ValidateAttribute implements attribute value validation. This type requires the value to be a String value that
+// is valid RFC 3339 format. This utilizes the Go `time` library which does not strictly adhere to the RFC 3339
+// standard and may allow strings that are not valid RFC 3339 strings
+//
+// See https://github.com/golang/go/issues/54580 for more info on the Go `time` library's RFC 3339 parsing differences.
+func (v RFC3339) ValidateAttribute(ctx context.Context, req xattr.ValidateAttributeRequest, resp *xattr.ValidateAttributeResponse) {
+	if v.IsUnknown() || v.IsNull() {
+		return
+	}
+
+	if _, err := time.Parse(time.RFC3339, v.ValueString()); err != nil {
+		resp.Diagnostics.Append(diag.WithPath(req.Path, rfc3339InvalidStringDiagnostic(v.ValueString(), err)))
```

Comment:
Super nit: Should we just move the now-single-use diagnostic code here so it can just be:

```suggestion
		resp.Diagnostics.Append(
			diag.NewAttributeErrorDiagnostic(
				req.Path,
				"...",
				"...",
			),
		)
```

Reply:
It looks like there are a couple of uses of `rfc3339InvalidStringDiagnostic()`:

- https://github.com/hashicorp/terraform-plugin-framework-timetypes/pull/58/files#diff-f8ecd86c8d3634e21eeb453046f07ee31969c2c8296c343fdd8f9d506ed9910aR94
- https://github.com/hashicorp/terraform-plugin-framework-timetypes/pull/58/files#diff-f8ecd86c8d3634e21eeb453046f07ee31969c2c8296c343fdd8f9d506ed9910aL143

I was thinking to leave this as-is, to keep the diagnostic code DRY, but happy to refactor if you think otherwise.